Can't pipe

Do you expect users to be able to pipe?

``` sh
$ echo {} | /usr/local/lib/node_modules/cdir/bin/cdir

node.js:201
        throw e; // process.nextTick error, or 'error' event on first tick
              ^
AssertionError: stdin must be initialized before calling setRawMode
    at Object.setRawMode (tty.js:37:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/cdir/cdir.js:7:5)
    at Module._compile (module.js:432:26)
    at Object..js (module.js:450:10)
    at Module.load (module.js:351:31)
    at Function._load (module.js:310:12)
    at Module.require (module.js:357:17)
    at require (module.js:368:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/cdir/bin/cdir:4:15)
    at Module._compile (module.js:432:26)
```

--- a/bin/cdir
+++ b/bin/cdir
@@ -1,12 +1,34 @@
 #!/usr/bin/env node
 
 var fs = require('fs');
-console.dir = require('../cdir');
+var vm = require('vm');
 
 var filename = process.argv[2];
+var subject = '';
 
-if (!filename) {
-  throw new Error('Requires a filename');
+if (filename) {
+  // read the contents of the given filename
+  fs.readFile(filename, 'utf8', function(err, file) {
+    if (err) throw err;
+    subject = file;
+    cdir();
+  });
+} else {
+  // read the contents of stdin
+  process.stdin.setEncoding('utf8');
+  process.stdin.on('data', function(data) {
+    subject += data;
+  });
+  process.stdin.on('end', function() {
+    cdir();
+  });
+  process.stdin.resume();
+  filename = '[stdin]';
 }
 
-console.dir(JSON.parse(fs.readFileSync(filename)));
+function cdir() {
+  var sandbox = {};
+  subject = vm.runInNewContext('(' + subject + ')', sandbox, filename);
+  console.dir = require('../cdir');
+  console.dir(subject);
+}

--- a/cdir.js
+++ b/cdir.js
@@ -1,13 +1,15 @@
 var tty = require('tty');
+var ttys = require('ttys');
 var rl = require('readline');
 
-var stdin;
+var stdin = ttys.stdin;
+var stdout = ttys.stdout;
 
 //
 // shorthand write to stdout
 //
 var write = function write (s) {
-  process.stdout.write(s);
+  stdout.write(s);
 }
 
 //
@@ -132,7 +134,6 @@ if (typeof JSON.decycle !== 'function') {
 //
 module.exports = function dir (obj, options) {
 
-  var stdin;
   var displayed = 0;
   var copybuffer = 0;
 
@@ -170,7 +171,7 @@ module.exports = function dir (obj, options) {
 
       var buffer = '';
       var description = ws(indent, true);
-      var maxWidth = process.stdout.getWindowSize()[0] - indent - 6;
+      var maxWidth = stdout.getWindowSize()[0] - indent - 6;
 
       for (var i = 0, cpos = 0, l = str.length; i < l; i++, cpos++) {
 
@@ -217,8 +218,8 @@ module.exports = function dir (obj, options) {
         var truncatedNode = '0';
         var truncated = false;
 
-          if (node.length > process.stdout.getWindowSize()[0] - extLen) {
-            truncatedNode = '▸ ' + '\033[31m"' + node.substr(0, process.stdout.getWindowSize()[0]/2) + '..."\033[0m';
+          if (node.length > stdout.getWindowSize()[0] - extLen) {
+            truncatedNode = '▸ ' + '\033[31m"' + node.substr(0, stdout.getWindowSize()[0]/2) + '..."\033[0m';
             truncated = true;
           }
           else {
@@ -444,10 +445,10 @@ module.exports = function dir (obj, options) {
       // include that before the prompt as the default.
       //
       if (lastsearch !== '') {
-        process.stdout.write('(' + lastsearch + ') /');
+        stdout.write('(' + lastsearch + ') /');
       }
       else {
-        process.stdout.write('/');
+        stdout.write('/');
       }
     }
     else if (searchmode === true && typeof key !== 'undefined' && key.name === 'backspace') {
@@ -652,7 +653,6 @@ module.exports = function dir (obj, options) {
     setRawMode.call(stdin, mode);
   }
 
-  stdin = process.stdin;
   stdin.resume();
 
   var wasRaw;

--- a/package.json
+++ b/package.json
@@ -21,6 +21,9 @@
   "bin": {
     "cdir": "./bin/cdir"
   },
+  "dependencies": {
+    "ttys": "*"
+  },
   "engines" : ["node >= 0.6.x"],
   "main" : "./cdir",
   "scripts": { 

